File: Atividades A1/15 Questões/Java/Questoes15/src/Questao14/Main.java
package Questao14;
//14. Singleton Implemente o padrão de projeto Singleton para garantir que apenas uma instância de uma classe Configuracao seja criada.

class Configuracao {
    private static Configuracao instance;

    private Configuracao() {
        System.out.println("Nova instância de Configuracao criada.");
    }

    public static Configuracao getInstance() {
        if (instance == null) instance = new Configuracao();

        return instance;
    }

    public void mostrarConfiguracao() {
        System.out.println("Exibindo configurações...");
    }
}

public class Main {
    public static void main(String[] args) {
        Configuracao config1 = Configuracao.getInstance();
        Configuracao config2 = Configuracao.getInstance();

        System.out.println(config1 == config2);

        config1.mostrarConfiguracao();
    }
}

File: Atividades A1/15 Questões/Python/Questoes15/Questao14/Main.py
#14. Singleton Implemente o padrão de projeto Singleton para garantir que apenas uma instância de uma classe Configuracao seja criada.

class Configuracao:
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(Configuracao, cls).__new__(cls)
            print('Nova instância de Configuracao criada.')
        return cls._instance

    def mostrarConfiguracao(self):
        print('Exibindo configurações...')


if __name__ == '__main__':
    config1 = Configuracao()
    config2 = Configuracao()

    print(config1 is config2)

    config1.mostrarConfiguracao()